src/devtools/client/webconsole/components/Output/ConsoleOutput.tsx

```diff
@@ -232,11 +366,44 @@ function mapStateToProps(state: UIState) {
     visibleMessages: selectors.getVisibleMessages(state),
   };
 }
+
 const mapDispatchToProps = (dispatch: Dispatch) => ({
   dispatch,
-  openLink: actions.openLink,
 });
 
 const connector = connect(mapStateToProps, mapDispatchToProps);
 
 export default connector(ConsoleOutput);
+
+// TODO Is there a more central location for this utility method?
+async function getTimeForMessage(message: Message): Promise<number> {
+  const { executionPoint, executionPointTime, timeStamp } = message;
+  if (timeStamp != null) {
+    return timeStamp;
+  } else if (executionPointTime != null) {
+    return executionPointTime;
+  } else if (executionPoint != null) {
+    return new Promise(async resolve => {
+      await analysisManager.runAnalysis(
+        {
+          effectful: true,
```

Comment:
you should be able to get away with passing `false` here since we are guaranteed not to be mutating anything I think?

Reply:
Ah! I thought you said it should be `true` because of potential diverging. But I can change it. I don't know what it should be given the docs :D

Reply:
Sorry! I was probably confusing true/false for `effectful` and `pure`, since they are opposites 😵

Reply:
Okay. `effectful: false` makes more sense to me 👍🏼